api/gitlab.ts

```diff
@@ -0,0 +1,269 @@
+        color: 'blue'
+      }
+    case 'open-issues':
+      return {
+        subject: 'open issues',
+        status: millify(result.openIssuesCount),
+        color: result.openIssuesCount === 0 ? 'green' : 'orange'
+      }
+    case 'closed-issues':
+      return {
+        subject: 'closed issues',
+        status: millify(result.issues.count),
+        color: 'blue'
+      }
+    case 'label-issues':
+      return {
+        subject: `${restArgs.label}`,
+        status: result.label ? millify(result.issues.count) : '0',
+        color: result.label ? removeNoSignFromHexColor(result.label.color) : 'gray'
+      }
+    default:
+      return {
+        subject: 'gitlab',
+        status: 'unknown topic',
+        color: 'grey'
+      }
+  }
+}
+
+const makeQueryCall = async ({ topic, owner, repo, ...restArgs }) => {
+  const repoQueryBodies = {
+    'stars': 'starCount',
+    'forks': 'forksCount',
+    'open-issues': 'openIssuesCount',
+    'closed-issues': 'issues(state:closed){ count }',
+    'issues': 'issues{ count }',
+  }
+
+  let queryBody
+  switch (topic) {
+    case 'label-issues':
+      const { label, state } = restArgs
+      const stateFilter = state ? `state:${state.toLowerCase()}` : ''
+      queryBody = `
+      issues(labelName:"${label}", ${stateFilter}) { count }
+      label(title: "${label}"){ color }
+      `
+      break
+    default:
+      queryBody = repoQueryBodies[topic]
+  }
+
+  const query = `
+  query {
+    project(fullPath:"${owner}/${repo}") {
+      ${queryBody}
+    }
+  }
+`
+  return queryGitlab(query).then(res => res.data!.project)
+}
+
+const makeRestCall = async ({ topic, owner, repo, ...restArgs }) => {
+  const restPaths = {
+    'mrs': `/projects/${encodeURIComponent(`${owner}/${repo}`)}/merge_requests`,
+    'open-mrs': `/projects/${encodeURIComponent(`${owner}/${repo}`)}/merge_requests?state=opened`,
+    'closed-mrs': `/projects/${encodeURIComponent(`${owner}/${repo}`)}/merge_requests?state=closed`,
+    'merged-mrs': `/projects/${encodeURIComponent(`${owner}/${repo}`)}/merge_requests?state=merged`,
+    'commits': `/projects/${encodeURIComponent(`${owner}/${repo}`)}/repository/commits?${restArgs.ref ? "ref_name=" + restArgs.ref : ''}`,
+    'last-commit': `/projects/${encodeURIComponent(`${owner}/${repo}`)}/repository/commits?${restArgs.ref ? "ref_name=" + restArgs.ref : ''}`,
+    'branches': `/projects/${encodeURIComponent(`${owner}/${repo}`)}/repository/branches`,
+    'tags': `/projects/${encodeURIComponent(`${owner}/${repo}`)}/repository/tags`,
+    'contributors': `/projects/${encodeURIComponent(`${owner}/${repo}`)}/repository/contributors`,
+    'releases': `/projects/${encodeURIComponent(`${owner}/${repo}`)}/releases`,
+    'release': `/projects/${encodeURIComponent(`${owner}/${repo}`)}/releases`,
+    'license': `/projects/${encodeURIComponent(`${owner}/${repo}`)}?license=true`,
+  }
+
+  let restPath
+  switch (topic) {
+    default:
+      restPath = restPaths[topic]
+  }
```

Comment:
Seems this `switch` is not needed anymore

Reply:
Yup. I'll remove that too